python/cudf/cudf/core/dtypes.py

```diff
@@ -982,6 +980,13 @@ def __init__(
             dtypes = {}
         else:
             self._subtype = cudf.dtype(subtype)
+            if isinstance(
+                self._subtype, cudf.CategoricalDtype
+            ) or cudf.utils.dtypes.is_dtype_obj_string(self._subtype):
+                raise TypeError(
+                    "category, object, and string subtypes are not supported "
+                    "for IntervalDtype"
+                )
```

Comment:
I imagine this should be in `cudf.dtype` instead

Reply:
Shouldn't it be here because all code-paths `IntervalDtype.from_arrow`, `.from_pandas` and pickle/unpickling go through this constructor?

Reply:
The thinking is similar to my comment in https://github.com/rapidsai/cudf/pull/19478/files#r2237928703 where if we disallow constructing this type here, we should ideally not allow constructing this type anywhere else for consistency.

But not a blocker for this PR